src/features/device-proxy/device-proxy.ts

```diff
@@ -214,22 +243,20 @@ async function requestDevices({
 			}
 			throw new NotFoundError('No online device(s) found');
 		}
-		// Check for device update permission, except for
-		// internal operation of the platform.
+		// Check resource access, unless request is for internal operations of the platform.
 		if (method !== 'GET' && req !== permissions.root) {
-			await Promise.all(
-				deviceIds.map(async (deviceId) => {
-					const res = (await resinApi.request({
-						method: 'POST',
-						url: `device(${deviceId})/canAccess`,
-						body: { action: 'update' },
-					})) as { d?: Array<{ id: number }> };
-
-					if (res?.d?.[0]?.id !== deviceId) {
-						throw new errors.ForbiddenError();
-					}
-				}),
-			);
+			const permission = ['PUT', 'PATCH', 'POST', 'DELETE'].includes(method)
+				? 'supervisor-proxy-write'
+				: 'update';
```

Comment:
Isn't this covering all cases really? I mean, unless I'm forgetting something, I would expect `['PUT', 'PATCH', 'POST', 'DELETE'].includes(method)` to be always true, so we could just do:
```suggestion
			const permission = 'supervisor-proxy-write';
```

Reply:
Yeah, it was about `HEAD` requests. If it's fine to not check `/canAccess` for `HEAD` requests then I'd prefer to go with your suggestion and add it next to `method !== 'GET'`.

Reply:
I think so, especially given that the supervisor doesn't support any HEAD requests atm afaict https://docs.balena.io/reference/supervisor/supervisor-api/